napari/_qt/dialogs/screenshot_dialog.py

```diff
@@ -62,8 +62,10 @@ def accept(self):
                     QMessageBox.No,
                 )
                 if res != QMessageBox.Yes:
-                    # standard accept return 1, reject 0. This inform that dialog should be reopened
-                    super().accept()
-                    self.exec_()
-        self.save_function(save_path)
-        return super().accept()
+                    # return in this case since still a valida name for the
+                    # file is needed and the dialog need to still be visible
+                    return None
+        base_result = super().accept()
+        if base_result:
+            self.save_function(save_path)
+        return base_result
```

Comment:
Wait. It looks like I was wrong here. Base on Qt documentation the accept() should return `None` (void in c++) https://doc.qt.io/qt-6/qdialog.html#accept 

```suggestion
        self.save_function(save_path)
        return base_result
```

If we would like to validate then we should depend on `self.result()`

Reply:
Oh totally right! So the validation should be something like: `if self.result() == QDialog.Accepted:` maybe?

Reply:
maybe just `if self.result()`